spanner/jdbc/src/main/java/com/example/spanner/JdbcSample.java

```diff
@@ -0,0 +1,214 @@
+import java.sql.Types;
+import java.util.Arrays;
+import java.util.List;
+
+/** Example showing how to use the Cloud Spanner open source JDBC driver. */
+public class JdbcSample {
+
+  /** Class to contain singer sample data. */
+  static class Singer {
+
+    final long singerId;
+    final String firstName;
+    final String lastName;
+
+    Singer(long singerId, String firstName, String lastName) {
+      this.singerId = singerId;
+      this.firstName = firstName;
+      this.lastName = lastName;
+    }
+  }
+
+  static final List<Singer> SINGERS =
+      Arrays.asList(
+          new Singer(1, "Marc", "Richards"),
+          new Singer(2, "Catalina", "Smith"),
+          new Singer(3, "Alice", "Trentor"),
+          new Singer(4, "Lea", "Martin"),
+          new Singer(5, "David", "Lomond"));
+
+  /**
+   * This example shows how to create a JDBC connection and use this to execute queries and updates.
+   */
+  public static void main(String[] args) throws Exception {
+    if (args.length != 3) {
+      printUsageAndExit();
+    }
+
+    /*
+     * CREATE TABLE Singers (
+     *   SingerId   INT64 NOT NULL,
+     *   FirstName  STRING(1024),
+     *   LastName   STRING(1024),
+     *   SingerInfo BYTES(MAX),
+     * ) PRIMARY KEY (SingerId);
+     */
+
+    String command = args[0];
+    String projectId = SpannerOptions.getDefaultProjectId();
+    String instanceId = args[1];
+    String databaseId = args[2];
+
+    String connectionUrl =
+        String.format(
+            "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s",
+            projectId, instanceId, databaseId);
+    try (Connection connection = DriverManager.getConnection(connectionUrl)) {
+      run(command, connection, projectId, instanceId, databaseId);
+    }
+  }
+
+  static void run(
+      String command, Connection connection, String projectId, String instanceId, String databaseId)
+      throws SQLException {
+    switch (command) {
+      case "createtable":
+        createTable(connection);
+        break;
+      case "insertdata":
+        insertData(connection);
+        break;
+      case "connectionwithqueryoptions":
+        connectionWithQueryOptions(projectId, instanceId, databaseId);
+        break;
+      case "setqueryoptions":
+        setQueryOptions(projectId, instanceId, databaseId);
+        break;
+      default:
+        printUsageAndExit();
+    }
+  }
+
+  static void printUsageAndExit() {
+    System.err.println("Usage:");
+    System.err.println("    JdbcSample <command> <instance_id> <database_id>");
+    System.err.println("");
+    System.err.println("Examples:");
+    System.err.println("    JdbcSample createtable my-instance example-db");
+    System.err.println("    JdbcSample insertdata my-instance example-db");
+    System.err.println("    JdbcSample connectionwithqueryoptions my-instance example-db");
+    System.err.println("    JdbcSample setqueryoptions my-instance example-db");
+
+    System.exit(1);
+  }
```

Comment:
I think we try to avoid using CLI tools for java samples.  I'll approve this since I couldn't find a prohibition in https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md

Reply:
This is modeled to work in the same way as the existing `SpannerSample` samples: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/c3c4600b0848a48dece316cf809f95721c427f44/spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java#L1597-L1610